test/fixtures/is-plain-object/case-1/after.js

```diff
@@ -0,0 +1,20 @@
+import assert from 'assert';
+
+// Test cases
+assert.strictEqual((typeof Object.create({}) === "object" && (Object.create({}).constructor === Object || Object.create({}).constructor === undefined)), true);
+assert.strictEqual((typeof Object.create(Object.prototype) === "object" && (Object.create(Object.prototype).constructor === Object || Object.create(Object.prototype).constructor === undefined)), true);
+assert.strictEqual((typeof { foo: 'bar' } === "object" && ({ foo: 'bar' }.constructor === Object || { foo: 'bar' }.constructor === undefined)), true);
+assert.strictEqual((typeof {} === "object" && ({}.constructor === Object || {}.constructor === undefined)), true);
+assert.strictEqual((typeof null === "object" && null !== null && (null.constructor === Object || null.constructor === undefined)), false);
+
+assert.strictEqual((typeof 1 === "object" && 1 !== null && ((1).constructor === Object || (1).constructor === undefined)), false);
+assert.strictEqual((typeof ['foo', 'bar'] === "object" && (['foo', 'bar'].constructor === Object || ['foo', 'bar'].constructor === undefined)), false);
+assert.strictEqual((typeof [] === "object" && ([].constructor === Object || [].constructor === undefined)), false);
+
+const Foo = function () { };
+assert.strictEqual((typeof new Foo() === "object" && (new Foo().constructor === Object || new Foo().constructor === undefined)), false);
```

Comment:
`new Foo()` and other cases should be called only once -> cached since this could change state and be heavy.

Reply:
good catch!